### PR TITLE
Make BindGroupDescriptor look like BindGroupLayoutDescriptor

### PR DESCRIPTION
**Pull request by Kangz**
PTAL!

### CODE REVIEW

File: design/sketch.webidl [hunk 1]
```diff
@@ -212,18 +212,17 @@ interface WebGPUPipelineLayout {
 };
 
 // BindGroup
-dictionary WebGPUBufferBinding {
+interface WebGPUBufferBinding {
```
Comment (kvark) [inline design/sketch.webidl]: why is this an interface?

File: design/sketch.webidl [hunk 2]
```diff
@@ -190,14 +190,14 @@ enum WebGPUBindingType {
...
-dictionary WebGPUBindGroupBinding {
+dictionary WebGPUBindGroupLayoutBinding {
     u32 binding;
     WebGPUShaderStageFlags visibility;
     WebGPUBindingType type;
 };
 
 dictionary WebGPUBindGroupLayoutDescriptor {
-    sequence<WebGPUBindGroupBinding> bindingTypes;
+    sequence<WebGPUBindGroupLayoutBinding> bindingTypes;
```
Comment (kvark) [inline design/sketch.webidl]: let's rename the field to `bindings`

File: design/sketch.webidl [hunk 3]
```diff
@@ -212,18 +212,17 @@ interface WebGPUPipelineLayout {
...
     u32 size;
 };
 
 typedef (WebGPUSampler or WebGPUTextureView or WebGPUBufferBinding) WebGPUBindingResource;
 
-dictionary WebGPUBinding {
-    sequence<WebGPUBindingResource> resources;
-    u32 start;
-    u32 count;
+interface WebGPUBinding {
```
Comment (kvark) [inline design/sketch.webidl]: same here

File: design/sketch.webidl (hunk 1 above)
Comment (Kangz) [reply design/sketch.webidl]: I'm not sure what the difference is tbh. @litherum seemed to say WebGPUBinding should be an interface so I made this one an interface too.

File: design/sketch.webidl (hunk 2 above)
Comment (Kangz) [reply design/sketch.webidl]: Done

File: design/sketch.webidl (hunk 3 above)
Comment (Kangz) [reply design/sketch.webidl]: See comment above.

**Comment by Kangz** [comment]
Ok, changed it to `dictionary`

**Comment by Kangz** [comment]
Thanks!

**Comment by kainino0x** [comment]
We'll need some clarification from @litherum on the `interface` topic. I don't see why we would use interfaces here at all.

**Comment by kainino0x** [comment]
IIUC, https://heycam.github.io/webidl/#es-interface says an `interface` can only be implemented by a platform object - it can't be a JS object (`{}`). (A user object can only be used as a `callback interface`.)

**Pull request merged**